lib/resty/core/regex.lua

```diff
@@ -153,6 +154,14 @@ local function get_max_regex_cache_size()
     return max_regex_cache_size
 end
 
+function _M.regex_cache_is_empty()
+    if not regex_match_cache then
+        return true
+    else
+        return next(regex_match_cache.hasht) == nil
```

Comment:
Also, directly indexing internal fields of `resty.lrucache` instances break encapsulation and poses risk of future incompatibilities.